addOns/ascanrules/src/test/java/org/zaproxy/zap/extension/ascanrules/HiddenFilesScanRuleUnitTest.java

```diff
@@ -86,6 +87,29 @@ void shouldHavePayloadsFile() {
         assertThat(url, is(notNullValue()));
     }
 
+    @Test
+    void checkNoPathsHaveLeadingSlash() {
+        // Given
+        URL url =
+                HiddenFilesScanRule.class.getResource("/" + HiddenFilesScanRule.PAYLOADS_FILE_PATH);
+        // When
+        List<HiddenFile> files = rule.readFromJsonFile(url.getPath());
+        // Then
+        // Check json file entries
+        assertThat(files.size(), is(greaterThan(1)));
+        for (HiddenFile file : files) {
+            String path = file.getPath();
+            assertThat(path + " should not start with slash (/).", !path.startsWith("/"), is(true));
```

Comment:
Worth extracting a method, to have the constraint being checked in a single place.